src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+  }
+
+  private final void classifyIfActiveThisMonth(RunDto run, SampleDto stock, ReportItem match, Map<String, ReportItem> thisMonth) {
+    if ((RUN_COMPLETED.equals(run.getState())
+        && run.getCompletionDate().compareTo(start) > 0
+        && run.getCompletionDate().compareTo(end) < 0)
+        ||
+        (RUN_RUNNING.equals(run.getState()) &&
+            (run.getStartDate().compareTo(start) > 0
+                && run.getStartDate().compareTo(end) < 0))) {
+      thisMonth.put(stock.getId(), match);
+    }
+  }
+
+  private static final List<Map.Entry<String, ReportItem>> listifyMap(Map<String, ReportItem> map) {
+    // need to convert it to a list, because getRow() calls item by index
+    ArrayList<Map.Entry<String, ReportItem>> listified = new ArrayList<>(map.entrySet());
+    Collections.sort(listified, new Comparator<Map.Entry<String, ReportItem>>() {
+      @Override
+      public int compare(Map.Entry<String, ReportItem> s1, Map.Entry<String, ReportItem> s2) {
+        return s1.getValue().getStock().getName().compareTo(s2.getValue().getStock().getName());
+      }
+    });
+    return listified;
+  }
+
+  private static boolean hasDnaAnalyteName(SampleDto dto) {
+    return (dto.getName().endsWith("_D") || dto.getName().contains("_D_"))
+        && !dto.getSampleType().contains("cDNA");
+  }
+
+  private Predicate<RunDto> byCreatedBefore(String end) {
+    return dto -> {
+      return start == null && end != null || dto.getCreatedDate().compareTo(end) < 0;
+    };
+  }
+
+  @Override
+  protected List<ColumnDefinition> getColumns() {
+    String analyteHeader = (analyte == null ? "DNA" : analyte);
+    return Arrays.asList(
+        new ColumnDefinition(String.format("%s: %s Recent (%s - %s)", project, analyteHeader, start, end)),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""));
+  }
+
+  @Override
+  protected int getRowCount() {
+    if (analyte == null) {
+      // print DNA and RNA on same sheet
+      return dnaRecentList.size()
+          + 3
+          + rnaRecentList.size()
+          + 3
+          + dnaAllList.size()
+          + 3
+          + rnaAllList.size()
+          + 1;
+    } else if (DNA.equals(analyte)) {
+      // only DNA
+      return dnaRecentList.size()
+          + 3
+          + dnaAllList.size()
+          + 1;
+    } else {
+      // only RNA
+      return rnaRecentList.size()
+          + 3
+          + rnaAllList.size()
+          + 1;
+    }
+    // the "+1" is to make the getRow() call happy -- adds a blank row at the end since
+    // don't know whether the last row will be DNA or RNA
+  }
+
+  @Override
+  protected String[] getRow(int rowNum) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)